pandas/core/algorithms.py

```diff
@@ -409,15 +414,14 @@ def isin(comps, values) -> np.ndarray:
             f"to isin(), you passed a [{type(values).__name__}]"
         )
 
-    if not isinstance(values, (ABCIndex, ABCSeries, np.ndarray)):
+    if not isinstance(values, (ABCIndex, ABCSeries, ABCExtensionArray, np.ndarray)):
         values = construct_1d_object_array_from_listlike(list(values))
 
+    comps = extract_array(comps, extract_numpy=True)
     if is_categorical_dtype(comps):
         # TODO(extension)
         # handle categoricals
-        return comps._values.isin(values)
-
-    comps = com.values_from_object(comps)
+        return comps.isin(values)  # type: ignore
```

Comment:
why is this type ignore?

Reply:
bc mypy doesnt know enough to rule out ndarray (or even non-Categorical EA) here

Reply:
not sure i understand. shouldn't you just add

```
result = comps.isin(values)
assert isinstance(result, np.ndarray)
return result
``` 

prefer this to a type ignore

Reply:
the issue isnt `type(result)`, its `type(comps)`.  `Categorical` has a `isin` method, but general ExtensionArray does not

Reply:
ok, is there a way to avoid the type ignore, i hate these things because we never remove them

Reply:
> i hate these things because we never remove them

yah, i keep a list of patterns to regularly grep for and try to get rid of.

> is there a way to avoid the type ignore

AFAIK the only way to do that would be to do a runtime import of Categorical and `assert isinstance(comps, Categorical)` which I think is more code-smelly than the "type: ignore".  eventually numpy will have stubs so mypy wont resolve np.ndarray to Any, at which point we'll have an opportunity to improve this

Reply:
what about an `isinstance(ABCCategorical) ` ?

Reply:
or ok with a run-time import of Categorical if the above doesn't work;

Reply:
mypy doesnt recognize any of our ABCFoos

Reply:
so you have an aversion to "# type: ignore" and i have an aversion to a runtime import just for mypy.  is there a middle ground somewhere?  maybe revisit in a few months to see if the type ignore can be removed?

Reply:
ok, let's do that, also let's create an issue to see if we can fix the ABC's and mypy